QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs

```diff
@@ -0,0 +1,175 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using QuikForm.Entities;
+using QuikForm.Repositories.Contexts;
+using QuikForm.Repositories.Repositories;
+using QuikForm.Repository.Contracts.Exceptions.Questions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QuikForm.Repositories.Repositories.Tests;
+
+[TestClass()]
+public class QuestionRepositoryTests
+{
+    #region CreateAsync
+    [TestMethod()]
+    public async Task CreateAsync_QuestionToAdd_QuestionWithId()
+    {
+        // Arrange
+        DbContextOptionsBuilder<ApplicationDbContext> builder = new DbContextOptionsBuilder<ApplicationDbContext>().UseInMemoryDatabase("QuikFormTest");
+        ApplicationDbContext context = new ApplicationDbContext(builder.Options);
+        context.Database.EnsureDeleted();
+        QuestionRepository questionRepository = new QuestionRepository(context);
+        int idExpected = 1;
+        Question questionToAdd = new Question() { Label = "Label" };
+
+        // Act
+        await questionRepository.CreateAsync(questionToAdd);
+
+        // Assert
+        Assert.AreEqual(idExpected, context.Questions.Find(1)?.Id);
+    }
+    #endregion
+
+    #region DeleteAsync
+    [TestMethod()]
+    public async Task DeleteAsync_QuestionToDelete_EmptyList()
+    {
+        // Arrange
+        DbContextOptionsBuilder<ApplicationDbContext> builder = new DbContextOptionsBuilder<ApplicationDbContext>().UseInMemoryDatabase("QuikFormTest");
+        ApplicationDbContext context = new ApplicationDbContext(builder.Options);
+        context.Database.EnsureDeleted();
+        QuestionRepository questionRepository = new QuestionRepository(context);
+        Question qToDelete = new Question() { Id = 1, Label = "Label" };
```

Comment:
questionToDelete or move in AddAsync inside

Reply:
i'm the best